### PR TITLE
FISH-315 Camel Case Testing

### PR DESCRIPTION
**Pull request by MattGill98**
Added testing for the camel casing function to make sure that notifier properties are calculated correctly the majority of the time. This is to ensure that notifier implementations don't see any unexpected results.

This PR wasn't necessary, but in implementing new notifiers I was getting anxiety at the sight of this method

### CODE REVIEW

**Comment by MattGill98** [comment]
jenkins test please

**Comment by MattGill98** [comment]
jenkins test please

File: nucleus/common/internal-api/src/test/java/fish/payara/internal/notification/CamelCaseConversionTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    public void if_trailing_symbols_expect_removed() {
+        assertEquals("abc", convertToCamelCase("abc_"));
+        assertEquals("abc", convertToCamelCase("abc&"));
+        assertEquals("abc", convertToCamelCase("abc-"));
+    }
+
+    @Test
+    public void if_internal_symbols_expect_next_character_capitalised() {
+        assertEquals("aBc", convertToCamelCase("a_bc"));
+        assertEquals("aBc", convertToCamelCase("a-bc"));
```
Comment (AlanRoth) [inline nucleus/common/internal-api/src/test/java/fish/payara/internal/notification/CamelCaseConversionTest.java:89]: Perhaps a assert for white space before or after a internal symbol

Comment (AlanRoth) [inline nucleus/common/internal-api/src/test/java/fish/payara/internal/notification/CamelCaseConversionTest.java:89]: Assert for multiple internal symbols?

File: nucleus/common/internal-api/src/test/java/fish/payara/internal/notification/CamelCaseConversionTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ * holder.
+ */
+package fish.payara.internal.notification;
+
+import static fish.payara.internal.notification.NotifierUtils.convertToCamelCase;
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class CamelCaseConversionTest {
```
Comment (AlanRoth) [inline nucleus/common/internal-api/src/test/java/fish/payara/internal/notification/CamelCaseConversionTest.java:47]: Could add a case for "   " expect "" or expect same, whichever is correct

`if (string == null || string.isEmpty()) {
            return string;
}`
Can do string.trim().isEmpty() and return string.trim(), so "" is returned if a whitespace string of any length is entered

File: nucleus/common/internal-api/src/test/java/fish/payara/internal/notification/CamelCaseConversionTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        assertEquals("abc", convertToCamelCase("abc "));
+        assertEquals("abc", convertToCamelCase("abc  "));
+        assertEquals("abc", convertToCamelCase(" abc"));
+        assertEquals("abc", convertToCamelCase("  abc"));
+    }
+
+    @Test
+    public void if_internal_whitespace_expect_next_character_capitalised() {
+        assertEquals("aBc", convertToCamelCase("a bc"));
+        assertEquals("abC", convertToCamelCase("ab c"));
```
Comment (AlanRoth) [inline nucleus/common/internal-api/src/test/java/fish/payara/internal/notification/CamelCaseConversionTest.java:103]: Assert for multiple whitespaces?

Comment (AlanRoth) [inline nucleus/common/internal-api/src/test/java/fish/payara/internal/notification/CamelCaseConversionTest.java:103]: Whats expected if there is a number followed by a whitespace and vice versa?

Comment (AlanRoth) [reply nucleus/common/internal-api/src/test/java/fish/payara/internal/notification/CamelCaseConversionTest.java:103]: Or number + internal symbol + whitespace?

**Review by cubastanley — APPROVED** [review]

**Review by AlanRoth — APPROVED** [review]

**Comment by MattGill98** [comment]
jenkins test please

**Comment by MattGill98** [comment]
jenkins test please

**Pull request merged**